doc: working_with_nrf: squash structure for nRF91 (#15139)
nrfconnect/sdk-nrf: doc/nrf/device_guides/nrf91/nrf91_programming.rst

.. _build_pgm_nrf9160:

Programming onto nRF91 Series devices
#####################################

.. contents::
   :local:
   :depth: 2

You can program applications and samples onto the nRF91 Series devices after :ref:`building` the corresponding firmware images.

> To program firmware onto nRF91 Series devices, follow the process described in the :ref:`programming` section, but pay attention to the exceptions specific to the nRF91 Series that are listed below.

Reviewer: ```suggestion
To program firmware onto nRF91 Series devices, follow the process described in the :ref:`programming` section.
Pay attention to the exceptions specific to the nRF91 Series, which are listed below.
```

.. |exceptions_step| replace:: Make sure to check the programming exceptions for the nRF91 Series in the sections below.

.. include:: /includes/vsc_build_and_run_series.txt

Compatible versions of modem and application
********************************************

When you update the application firmware on an nRF91 Series DK, make sure that the modem firmware and application firmware are compatible versions.

Selecting board controller on nRF9160 DK
****************************************

When programming on the nRF9160 DK, make sure to set the **SW10** switch (marked PROG/DEBUG) in the **nRF91** position to program the nRF9160 application.
In nRF9160 DK board revision v0.9.0 and earlier, the switch is called **SW5**.

.. _building_pgming:

Programming onto Thingy:91
**************************

> You can also program the Thingy:91 by using the images obtained by building the code in an |NCS| environment.

Reviewer: ```suggestion
You can also program the Thingy:91 with the images obtained by building the code in an |NCS| environment.
```

+-----------------------+----------------------------------------+----------------------------------------------------------------+
|:file:`app_update.bin` | MCUboot compatible image, binary format|* Using the built-in bootloader and mcumgr command line tool.   |
|                       |                                        |* For FOTA updates.                                             |
+-----------------------+----------------------------------------+----------------------------------------------------------------+

For an overview of different types of build files in the |NCS|, see :ref:`app_build_output_files`.

There are multiple methods of programming a sample or application onto a Thingy:91.
It is recommended to use an external debug probe to program the Thingy:91.

.. note::

   If you do not have an external debug probe available to program the Thingy:91, you can directly program by :ref:`using the USB (MCUboot) method and nRF Connect Programmer <programming_thingy>`.
   In this scenario, use the :file:`app_signed.hex` firmware image file.

.. note::

>    While building applications for Thingy:91, the build system changes the signing algorithm of MCUboot so that it uses the default RSA Keys.

Reviewer: ```suggestion
   While building applications for Thingy:91, the build system changes the signing algorithm of MCUboot so that it uses the default RSA keys.
```

   This is to ensure backward compatibility with the MCUboot versions that precede the |NCS| v1.4.0.

>    The default RSA keys must only be used for development.

Reviewer: ```suggestion
   Use the default RSA keys only for development.
```
